doc.js

```diff
@@ -220,6 +235,76 @@ define('doc', ['event'], function(event) {
 				return query(result);
 			},
 
+			'closest' : function(selector) {
+				// only works with non-composite selectors. e.g. '.class' or '#id' or 'div', but not: '#id .class' or 'ul.class'
+				var elements = [];
+				var selectorType;
+				var replacedSelector = selector;
+
+				if (matcher.isTag(selector)) {
+					selectorType = 'tag';
+				} else if (matcher.isClass(selector)) {
+					replacedSelector = selector.replace('.', '');
+					selectorType = 'class';
+				} else if (matcher.isId(selector)) {
+					replacedSelector = selector.replace('#', '');
+					selectorType = 'id';
+				} else {
+					console.warn('You are using composite selectors. e.g. "#id .class" or "tag.class". This selector will only work on Chrome 41+, Firefox 35+ or Opera 28+');
+				}
+
+				var checkForClosestParent = function(element) {
+					if (element) {
+						switch (selectorType) {
+							case 'tag':
+								return (replacedSelector.toUpperCase() === element.tagName)
+									? element
+									: checkForClosestParent(element.parentElement);
+								break;
+							case 'class':
+								return (query(element).hasClass(replacedSelector))
+									? element
+									: checkForClosestParent(element.parentElement);
+								break;
+							case 'id':
+								return (element.id === replacedSelector)
+									? element
+									: checkForClosestParent(element.parentElement);
+								break;
+							default:
+								throw new SyntaxError("You cannot use composite selector. e.g. 'tag.class' or '#id tag.class tag'. Use simple selectors like '#id', '.class' or 'tag'");
+								break;
+						}
+					}
+					return null;
+				};
+
+				for (var i = 0; i < this.size; i++) {
+					if (typeof this.els[i].closest === 'function') {
+						var el = this.els[i].closest(selector);
+						if (el) {
+							elements.push(el);
+						}
+					} else {
+						var closestParent = checkForClosestParent(this.els[i].parentElement);
+						if (closestParent) {
+							elements.push(closestParent);
+						}
+					}
+				}
+
+				for (var i = elements.length - 1; i > 0; i--) {
```

Comment:
Instead of looping again to remove duplicated items, you should verify if the element is already on the list before adding it.

@erikatakahara @williammizuta